packages/mobile/src/firebase/firebase.ts

```diff
@@ -218,39 +219,48 @@ export function appRemoteFeatureFlagChannel() {
-      emit({
-        hideVerification: flags?.hideVerification ?? false,
-        showRaiseDailyLimitTarget: flags?.showRaiseDailyLimitTargetV2 ?? undefined,
-        celoEducationUri: flags?.celoEducationUri ?? null,
-        celoEuroEnabled: flags?.celoEuroEnabled ?? false,
-        shortVerificationCodesEnabled: flags?.shortVerificationCodesEnabled ?? false,
-        inviteRewardsEnabled: flags?.inviteRewardsEnabled ?? false,
-        inviteRewardCusd: flags?.inviteRewardCusd ?? 1,
-        inviteRewardWeeklyLimit: flags?.inviteRewardWeeklyLimit ?? 5,
-        walletConnectEnabled: flags?.walletConnectEnabled ?? false,
-        rewardsABTestThreshold:
-          flags?.rewardsABTestThreshold ?? '0xffffffffffffffffffffffffffffffffffffffff',
-        rewardsPercent: flags?.rewardsPercent ?? 5,
-        rewardsStartDate: flags?.rewardsStartDate ?? 1622505600000,
-        rewardsMax: flags?.rewardsMax ?? 1000,
-      })
+    const emitter = (fetchedRemotely: boolean) => {
+      if (fetchedRemotely) {
+        Logger.debug('Configs were retrieved from the backend and activated.')
+        const flags: FirebaseRemoteConfigTypes.ConfigValues = remoteConfig().getAll()
+        emit({
+          hideVerification: flags.hideVerification.asBoolean(),
+          showRaiseDailyLimitTarget: flags.showRaiseDailyLimitTargetV2.asString(),
+          celoEducationUri: flags.celoEducationUri.asString(),
+          celoEuroEnabled: flags.celoEuroEnabled.asBoolean(),
+          shortVerificationCodesEnabled: flags.shortVerificationCodesEnabled.asBoolean(),
+          inviteRewardsEnabled: flags.inviteRewardsEnabled.asBoolean(),
+          // TODO: why was this defaulting to 1 when the initial state is 0?
+          inviteRewardCusd: flags.inviteRewardCusd.asNumber(),
+          // TODO: same here default 5 but initial state 0
+          inviteRewardWeeklyLimit: flags.inviteRewardWeeklyLimit.asNumber(),
+          walletConnectEnabled: flags.walletConnectEnabled.asBoolean(),
+          rewardsABTestThreshold: flags.rewardsABTestThreshold.asString(),
+          rewardsPercent: flags.rewardsPercent.asNumber(),
+          rewardsStartDate: flags.rewardsStartDate.asNumber(),
+          rewardsMax: flags.rewardsMax.asNumber(),
+        })
+      } else {
+        Logger.debug(
+          'No configs were fetched from the backend, and the local configs were already activated'
+        )
+      }
     }
 
-    const onValueChange = firebase
-      .database()
-      .ref('versions/flags')
-      .on(VALUE_CHANGE_HOOK, emitter, errorCallback)
+    const errorCallback = (error: Error) => {
+      Logger.warn(TAG, error.toString())
+    }
+
+    // Note: values are cached by default for 12 hours
+    // this can be changed if needed:
+    // https://rnfirebase.io/remote-config/usage
+    remoteConfig().fetchAndActivate().then(emitter).catch(errorCallback)
+
+    // TODO: is there an api for receiving updates without restart?
```

Comment:
Good catch, we just need to call `remoteConfig().fetchAndActivate()` again every X time. One option is doing it when the app is foregrounded.

Also, this is set as a channel emitter because it used to be able to get updates by itself. Now that it's just fetching once I don't think it makes much sense for it to be an emitter, it can just be an `async` function.

So, what I'd do is:

In the `appRemoteFeatureFlagSaga` (the place that is calling this), something like:
```
let time = Date.now()
while (true) {
  const appState: AppState = yield take(Actions.SET_APP_STATE) // Run the following when the app state is changed
  if (appState is active && Date.now() > time + 1_hour) {
    time = Date.now + 1_hour
    const flags: RemoteFeatureFlags | null = yield call(fetchRemoteFeatureFlags) // This is appRemoteFeatureFlagChannel renamed.
    if (flags) {
      yield put(updateFeatureFlags(flags))
    }
  }
}
```

And then change this function to an async function:
```
export async function fetchRemoteFeatureFlags() {
  if (!FIREBASE_ENABLED) { ... }
  const fetchedRemotely = await remoteConfig().fetchAndActivate()
  if (fetchedRemotely) {
    const flags: FirebaseRemoteConfigTypes.ConfigValues = remoteConfig().getAll()
    return {
      hideVerification: flags.hideVerification.asBoolean(),
      ...
    }
  } else {
    return null
  }
}
```

There's some tidying up and maybe logging needed in the previous code, but that's the general idea. 

Another option is instead of doing this when the app is opened doing it every one hour. The only way I know to do that is with `yield delay(60 * 60 * 1000)`, but I'm not sure what happens with long-running delays, which is why I didn't suggest it as the first option. There's probably a better way to run something every hour anyway 🤔

Reply:
+1 to @gnardini's solution.